src/mailsystem/postfix.cpp

```diff
@@ -50,3 +66,228 @@ void Postfix::setPassword(const std::optional<std::string>& password)
+}
+
+void Postfix::setDestination(
+    const std::optional<std::vector<std::string>>& destination)
+{
+    m_destination = destination;
+}
+
+const std::optional<std::string>& Postfix::getSMTPServer() const
+{
+    return m_smtp_server;
+}
+
+void Postfix::setSMTPServer(const std::optional<std::string>& smtp_server)
+{
+    m_smtp_server = smtp_server;
+}
+
+const std::optional<std::filesystem::path>& Postfix::getCertFile() const
+{
+    return m_cert_file;
+}
+
+void Postfix::setCertFile(const std::optional<std::filesystem::path>& cert_file)
+{
+    m_cert_file = cert_file;
+}
+
+const std::optional<std::filesystem::path>& Postfix::getKeyFile() const
+{
+    return m_key_file;
+}
+
+void Postfix::setKeyFile(const std::optional<std::filesystem::path>& key_file)
+{
+    m_key_file = key_file;
+}
+
+void Postfix::install()
+{
+    LOG_INFO("Installing Postfix")
+    // runCommand("dnf -y install postfix");
+}
+
+void Postfix::createFiles(const std::filesystem::path& basedir)
+{
+    LOG_INFO("Creating Postfix configuration files");
+    const std::filesystem::path mainFile = basedir / "main.cf";
+    const std::filesystem::path masterFile = basedir / "master.cf";
+    cloyster::removeFile(masterFile.string());
+
+    inifile baseini;
+    baseini.loadData(
+#include "cloysterhpc/tmpl/postfix/main.cf.tmpl"
+    );
+
+    if (m_hostname) {
+        baseini.setValue("", "myhostname", m_hostname.value());
+    }
+
+    if (m_domain) {
+        baseini.setValue("", "mydomain", m_domain.value());
+    }
+
+    if (m_destination) {
+        baseini.setValue("", "mydestination",
+            fmt::format("{}", fmt::join(m_destination.value(), ",")));
+    }
+
+    if (m_cert_file) {
+        baseini.setValue("", "smtpd_tls_cert_file", m_cert_file->string());
+    }
+
+    if (m_key_file) {
+        baseini.setValue("", "smtpd_tls_key_file", m_key_file->string());
+    }
+
+    inifile ini = baseini.mergeInto(mainFile);
+
+    //@TODO Check if m_domain is the right key. Maybe a new variable is needed
+    // here, m_relayhost_domain?.
+    switch (m_profile) {
+        case Profile::Local:
+            break;
+        case Profile::SASL:
+        case Profile::Relay:
+            if (m_domain && m_port) {
+                ini.setValue("", "relayhost",
+                    fmt::format("[{}]:{}", m_domain.value(), m_port.value()));
+            }
+            break;
+    }
+
+    ini.saveFile(mainFile);
+
+    const auto& masterConfInclude {
+#include "cloysterhpc/tmpl/postfix/master.cf.tmpl"
+    };
+
+    std::string masterConf(masterConfInclude);
+
+    if (m_profile != Profile::Local) {
+        masterConf
+            = cloyster::findAndReplace(masterConf, "{postfix_local_config}",
+                "#local     unix  -       n       n       -       -       ");
+    } else {
+        masterConf
+            = cloyster::findAndReplace(masterConf, "{postfix_local_config}",
+                "local     unix  -       n       n       -       -       ");
+    }
```

Comment:
To avoid this can we just change the line that matches local delivery? Why? To avoid adding a whole file as a template, so if anything changes on the RPM package it will not be an issue for us.